Feature part 4

JoeyAlanS opened the pull request.

JoeyAlanS committed:
### initial commit

JoeyAlanS committed:
###added
-added configuration
-added repositories
-added services

Bacuriim commented on src/main/java/com/eletra/controllers/CategoryController.java:
```diff
@@ -0,0 +1,64 @@
+        return list;
+    }
+
+    @PostMapping("/category")
+    @ResponseBody
+    @ApiOperation(value = "Return category")
+    public CategoryEntity postCategoryEntity(@RequestBody CategoryEntity categoryEntity) {
+        return categoryRepository.save(categoryEntity);
+    }
```
Acredito que o "Return category" nesse método seria um "New category" ou "Add category", pois o método postCategoryEntity adiciona uma nova categoria na lista.

Bacuriim commented on src/main/java/com/eletra/controllers/LineupController.java:
```diff
@@ -0,0 +1,57 @@
+        return lineupRepository.findByLineName(lineName);
+    }
+
+    @PostMapping("/line")
+    @ResponseBody
+    @ApiOperation(value = "Return line")
+    public LineupEntity postLineupEntity(@RequestBody LineupEntity lineupEntity) {
+        return lineupRepository.save(lineupEntity);
+    }
```
Mesmo ponto da categoryController, esse "Return line" deve ser alguma coisa que referencie uma adição, pois o método post adiciona um novo membro a lista.

Bacuriim commented on src/main/java/com/eletra/controllers/ModelController.java:
```diff
@@ -0,0 +1,62 @@
+        return list;
+    }
+
+    @PostMapping("/model")
+    @ResponseBody
+    @ApiOperation(value = "Return model")
+    public ModelEntity postModelEntity(@RequestBody ModelEntity modelEntity) {
+        return modelRepository.save(modelEntity);
+    }
```
Mesmo ponto da categoryController, esse "Return model" deve ser alguma coisa que referencie uma adição, pois o método post adiciona um novo membro a lista.

Bacuriim reviewed (changes requested):
Encontrei alguns poucos pontos no code review, por favor corrigi-los!

JoeyAlanS committed:
###Changed
-Changed postCategoryEntity
-Changed postLineupEntity
-Changed postModelEntity

Bacuriim reviewed (approved):
Não foram encontradas novas pendências!

gustavo-lopes-dev commented on src/main/java/com/eletra/Testando.java:
Por qual motivo a classe Main que inicializa a Sprint tem o nome "Testando"?

JoeyAlanS replied:
Havia esquecido de trocar o nome, por conta do tutorial.

gustavo-lopes-dev commented on src/main/java/com/eletra/controllers/CategoryController.java:
```diff
@@ -0,0 +1,64 @@
+        return "Category deleted";
+    }
+
+    @PutMapping("/category")
+    @ResponseBody
+    @ApiOperation(value = "Update category")
+    public String updateCategoryEntity(@RequestBody CategoryEntity categoryEntity) {
+        categoryRepository.save(categoryEntity);
+        return "Category updated";
+    }
```
Favor, não retornar uma String no método de update, e sim o proprio objeto salvo no banco, assim como é feito no método "postCategoryEntity"

gustavo-lopes-dev commented on src/main/java/com/eletra/controllers/CategoryController.java:
```diff
@@ -0,0 +1,64 @@
+        return categoryRepository.save(categoryEntity);
+    }
+
+    @DeleteMapping("/category/{category-name}")
+    @ResponseBody
+    @ApiOperation(value = "Delete category")
+    public String deleteCategoryEntity(@PathVariable(value = "category-name") String categoryName) {
+        CategoryEntity categoryEntity = categoryRepository.findByCategoryName(categoryName);
+        categoryRepository.delete(categoryEntity);
+        return "Category deleted";
+    }
```
Favor, melhorar retorno desse método, no caso ou retornar um valor booleano para retornar se realmente foi possível deletar esse item (porém, o método "delete" do JPA não retorna nada, logo seria interessante uma verificação se aquele item ainda continua no banco) ou então ser um método void

gustavo-lopes-dev commented on src/main/java/com/eletra/controllers/LineupController.java:
```diff
@@ -0,0 +1,57 @@
+        return lineupRepository.save(lineupEntity);
+    }
+
+    @DeleteMapping("/line/{line-name}")
+    @ResponseBody
+    @ApiOperation(value = "Delete line")
+    public String deleteLineupEntity(@PathVariable(value = "line-name") String lineName) {
+        LineupEntity lineupEntity = lineupRepository.findByLineName(lineName);
+        lineupRepository.delete(lineupEntity);
+        return "Line deleted";
+    }
```
Favor, melhorar retorno desse método, no caso ou retornar um valor booleano para retornar se realmente foi possível deletar esse item (porém, o método "delete" do JPA não retorna nada, logo seria interessante uma verificação se aquele item ainda continua no banco) ou então ser um método void

gustavo-lopes-dev commented on src/main/java/com/eletra/controllers/LineupController.java:
```diff
@@ -0,0 +1,57 @@
+        return "Line deleted";
+    }
+
+    @PutMapping("/line")
+    @ResponseBody
+    @ApiOperation(value = "Update line")
+    public String updateLineEntity(@RequestBody LineupEntity lineupEntity) {
+        lineupRepository.save(lineupEntity);
+        return "Line updated";
+    }
```
Favor, não retornar uma String no método de update, e sim o proprio objeto salvo no banco, assim como é feito no método "postLineEntity"

gustavo-lopes-dev commented on src/main/java/com/eletra/controllers/ModelController.java:
```diff
@@ -0,0 +1,62 @@
+        return "Model deleted";
+    }
+
+    @PutMapping("/model")
+    @ResponseBody
+    @ApiOperation(value = "Update model")
+    public String updateModelEntity(@RequestBody ModelEntity modelEntity) {
+        modelRepository.save(modelEntity);
+        return "Model updated";
+    }
```
Favor, não retornar uma String no método de update, e sim o proprio objeto salvo no banco, assim como é feito no método "postModelEntity"

gustavo-lopes-dev commented on src/main/java/com/eletra/controllers/ModelController.java:
```diff
@@ -0,0 +1,62 @@
+        return modelRepository.save(modelEntity);
+    }
+
+    @DeleteMapping("/model/{model-name}")
+    @ResponseBody
+    @ApiOperation(value = "Delete model")
+    public String deleteModelEntity(@PathVariable(value = "model-name") String modelName) {
+        ModelEntity modelEntity = modelRepository.findByModelName(modelName);
+        modelRepository.delete(modelEntity);
+        return "Model deleted";
+    }
```
Favor, melhorar retorno desse método, no caso ou retornar um valor booleano para retornar se realmente foi possível deletar esse item (porém, o método "delete" do JPA não retorna nada, logo seria interessante uma verificação se aquele item ainda continua no banco) ou então ser um método void

gustavo-lopes-dev reviewed (changes requested):
Foi realizado o code review e foram encontrados pontos de melhoria

JoeyAlanS committed:
###changed
- Changed methods CategoryController
- Changed methods LineupController
- Changed methods ModelController
- Changed name Testanto to Main

JoeyAlanS committed:
### Changed
-changed .ideaXML

JoeyAlanS committed:
###Changed
-Changed groupId on pom.xml

JoeyAlanS committed:
###Changed
-changed pom.xml

gustavo-lopes-dev reviewed (approved):
Não foram encontradas novas pendências
